adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -338,85 +334,179 @@ case class ReferenceRegion(
-    hull(region)
+  def merge(other: ReferenceRegion): ReferenceRegion = {
+    require(overlaps(other) || isAdjacent(other), "Cannot merge two regions that do not overlap or are not adjacent")
+    merge(other, 1L)
   }
 
   /**
-   * Calculates the intersection of two reference regions.
+   * Merges two reference regions that are within a threshold of each other.
    *
-   * @param region Region to intersect with.
-   * @return A smaller reference region.
+   * @throws IllegalArgumentException Thrown if regions are no within the distance threshold.
+   *
+   * @param other Other region to merge with this region.
+   * @return The merger of both unions.
+   *
+   * @see hull
    */
-  def intersection(region: ReferenceRegion): ReferenceRegion = {
-    assert(overlaps(region), "Cannot calculate the intersection of non-overlapping regions.")
-    ReferenceRegion(referenceName, max(start, region.start), min(end, region.end))
+  def merge(other: ReferenceRegion, distanceThreshold: Long): ReferenceRegion = {
+    require(isNearby(other, distanceThreshold), "Cannot merge two regions that do not meet the distance threshold")
+    require(distanceThreshold >= 0, "Distance must be non-negative number")
+    hull(other)
+  }
+
+  /**
+   * Calculates the intersection of two reference regions given a minimum
+   * overlap.
+   *
+   * @param other Region to intersect with.
+   * @param minOverlap The mimimum amount of positions that the two regions
+   *        must overlap by
+   * @return A smaller reference region
+   */
+  def intersection(other: ReferenceRegion, minOverlap: Long = 0L): ReferenceRegion = {
+    require(overlapsBy(other).exists(_ >= minOverlap), "Other region does not meet minimum overlap provided")
+    ReferenceRegion(referenceName, max(start, other.start), min(end, other.end), strand)
   }
 
   /**
    * Creates a region corresponding to the convex hull of two regions. Has no preconditions about the adjacency or
    * overlap of two regions. However, regions must be in the same reference space.
    *
-   * @throws AssertionError Thrown if regions are in different reference spaces.
+   * @throws IllegalArgumentException Thrown if regions are in different reference spaces.
    *
-   * @param region Other region to compute hull of with this region.
+   * @param other Other region to compute hull of with this region.
    * @return The convex hull of both unions.
    *
    * @see merge
    */
-  def hull(region: ReferenceRegion): ReferenceRegion = {
-    assert(strand == region.strand, "Cannot compute convex hull of differently oriented regions.")
-    assert(referenceName == region.referenceName, "Cannot compute convex hull of regions on different references.")
-    ReferenceRegion(referenceName, min(start, region.start), max(end, region.end))
+  def hull(other: ReferenceRegion): ReferenceRegion = {
+    require(sameStrand(other), "Cannot compute convex hull of differently oriented regions.")
```

Comment:
For require messages, use phrases or sentences or exclamations, but not all three :)